amr-wind/wind_energy/ABLFieldInit.cpp

```diff
@@ -76,10 +83,20 @@ void ABLFieldInit::operator()(
     const amrex::Real* th = m_thht_d.data();
     const amrex::Real* tv = m_thvv_d.data();
 
+    // Handle mesh mapping coordinates
+    Field const* nu_coord_cc =
+        m_mesh_mapping ? &(m_repo.get_field("non_uniform_coord_cc")) : nullptr;
+    amrex::Array4<amrex::Real const> nu_cc =
+        m_mesh_mapping ? ((*nu_coord_cc)(level).array(mfi))
+                       : amrex::Array4<amrex::Real const>();
+
     amrex::ParallelFor(vbx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-        const amrex::Real x = problo[0] + (i + 0.5) * dx[0];
-        const amrex::Real y = problo[1] + (j + 0.5) * dx[1];
-        const amrex::Real z = problo[2] + (k + 0.5) * dx[2];
+	const amrex::Real x = 
+            m_mesh_mapping ? nu_cc(i, j, k, 0) : problo[0] + (i + 0.5) * dx[0];
```

Comment:
Just from a reading standpoint, might help to put the `false` argument in parenthesis.